File: examples/java/10_interface/contapoupanca.java
public class ContaPoupanca impĺements Conta {
	private double saldo;
	
	public void depositar(double valor) {
		this.saldo += valor;
	}
	
	public double getSaldo() {
		return this.saldo;
	}
	
	public void sacar(double valor) {
		this.saldo -= valor;
	}
}

File: test/ebin/contapoupanca.erl
-module(contapoupanca).

-export([depositar/2, getSaldo/1, new/1, sacar/2]).

-implements(conta).

new(ValorInicial) ->
    ObjectID = ooe:new([{'Saldo', []}]),
    ooe:update_attr(ObjectID, 'Saldo', ValorInicial),
    {contapoupanca, ObjectID}.

depositar(ObjectID, Valor) ->
    ooe:update_attr(ObjectID, 'Saldo',
		    ooe:lookup_attr(ObjectID, 'Saldo') + Valor).

getSaldo(ObjectID) ->
    ooe:lookup_attr(ObjectID, 'Saldo').

sacar(ObjectID, Valor) ->
    ooe:update_attr(ObjectID, 'Saldo',
		    ooe:lookup_attr(ObjectID, 'Saldo') - Valor).
